packit/utils/obs_helper.py

```diff
@@ -0,0 +1,452 @@
+
+    toks.pop(0)  # Remove leading "*"
+    date = toks[:4]
+
+    del toks[:4]
+    for index, element in enumerate(toks):
+        if not element.startswith("<"):
+            author.append(element)
+        else:
+            email += element
+            del toks[: index + 1]
+            break
+
+    if toks and toks[0] == "-":
+        toks.pop(0)
+
+    if toks:
+        version = toks[0]
+
+    return date, author, email, version
+
+
+def format_changelog(changelog: list[str]) -> list[str]:
+    """Converts spec changelog to OBS .changes format"""
+    output: list[str] = []
+    changelog_time = "12:00:00 UTC"
+
+    # Remove leading newline
+    if changelog[0] == "\n":
+        changelog.pop(0)
+
+    for line in changelog:
+        if line.strip().startswith("*"):
+            date, author, email, version = parse_changelog_subject(line)
+
+            date.insert(3, changelog_time)
+            output.append("-" * 68)
+            output.append(f"{' '.join(date)} - {' '.join(author)} {email}")
+            output.append(f"- {version}")
+        else:
+            output.append(f"  {line}")
+
+    return output
+
+
+def create_changes_file(package_dir: Path):
+    """
+    Creates a changelog file from a by copying the changelog section from the spec file.
+    Sets release to 0 since obs handles release numbers
+
+    Args:
+        package_dir (Path): Path to the directory containing the package spec file.
+
+    Returns:
+        None
+    """
+    specfile: str
+    content: str
+    release_line = "Release:    0"
+
+    for file in list(filter(os.path.isfile, os.listdir(package_dir))):
+        if file.endswith(".spec"):
+            specfile = file
+            break
+
+        if file.endswith(".changes"):
+            logger.info(".changes file already exists in package")
+            return
+
+    if not specfile:
+        raise ValueError("Cannot find spec file in package")
+
+    with open(specfile) as s_file:
+        content = s_file.read()
+
+    split_content = re.split(r"%changelog", content, flags=re.IGNORECASE)
+    if len(split_content) < 2:
+        logger.info("Project has no changelog")
+        return
+
+    specfile_content, changes_file_content = split_content[0], "".join(
+        split_content[1:],
+    )
+
+    with open(package_dir / specfile, "w") as s_file:
+        specfile_content = "\n".join(
+            [
+                release_line if line.lower().startswith("release:") else line
+                for line in specfile_content.splitlines()
+            ],
+        )
+        s_file.write(specfile_content)
+
+    changes_filename = package_dir / f"{specfile[:-5]}.changes"
+
+    changes_file_content = "\n".join(
+        format_changelog(changes_file_content.splitlines()),
+    )
+
+    try:
+        with open(changes_filename, "w") as changes_file:
+            changes_file.write(changes_file_content)
+    except OSError as err:
+        logger.warn(f"Error writing {changes_filename} file: {err}")
```

Comment:
IMHO we shouldn't swallow exceptions here, this is a pretty fatal error. wdyt @lachmanfrantisek ?

```suggestion
    try:
        with open(changes_filename, "w") as changes_file:
            changes_file.write(changes_file_content)
    except OSError as err:
        logger.warn(f"Error writing {changes_filename} file: {err}")
```

Reply:
yes, it is quite unlikely to occur. If we were to have a file system error the build would fail long before this point.

Reply:
Then, we shouldn't catch this exception.